src/Features/Core/Portable/Completion/Providers/AbstractAwaitCompletionProvider.cs

```diff
@@ -91,18 +159,116 @@ private protected bool ShouldMakeContainerAsync(SyntaxToken token, SyntaxGenerat
+                        if (syntaxContextAtInsertationPosition.IsAwaitKeywordContext())
+                        {
+                            return IsConfigureAwaitable(syntaxContext.SemanticModel.Compilation, symbol)
+                                ? DotAwaitContext.AwaitAndConfigureAwait
+                                : DotAwaitContext.AwaitOnly;
+                        }
+                    }
+                }
+            }
+
+            return DotAwaitContext.None;
+        }
+
+        private IEnumerable<CompletionItem> GetCompletionItems(SyntaxToken token, bool isAwaitKeywordContext, DotAwaitContext dotAwaitContext, SyntaxGenerator generator, ISyntaxKindsService syntaxKinds, ISyntaxFactsService syntaxFacts)
         {
             var shouldMakeContainerAsync = ShouldMakeContainerAsync(token, generator);
-            var text = syntaxFacts.GetText(syntaxKinds.AwaitKeyword);
-            return CommonCompletionItem.Create(
-                displayText: text,
-                displayTextSuffix: "",
-                rules: CompletionItemRules.Default,
-                Glyph.Keyword,
-                description: RecommendedKeyword.CreateDisplayParts(text, FeaturesResources.Asynchronously_waits_for_the_task_to_finish),
-                inlineDescription: shouldMakeContainerAsync ? FeaturesResources.Make_containing_scope_async : null,
-                isComplexTextEdit: shouldMakeContainerAsync);
+            var displayText = syntaxFacts.GetText(syntaxKinds.AwaitKeyword);
+            var falseKeyword = syntaxFacts.GetText(syntaxKinds.FalseKeyword);
+            var filterText = displayText;
+            if (dotAwaitContext is DotAwaitContext.AwaitAndConfigureAwait)
+            {
+                // In the AwaitAndConfigureAwait case, we want to offer two completions: await and awaitf
+                // This case adds "await"
+                var completionPropertiesForAwaitOnly = GetCompletionProperties(token, isAwaitKeywordContext, DotAwaitContext.AwaitOnly, shouldMakeContainerAsync);
+                yield return CreateCompletionItem(displayText, filterText, falseKeyword, completionPropertiesForAwaitOnly);
+
+            }
+
+            var completionProperties = GetCompletionProperties(token, isAwaitKeywordContext, dotAwaitContext, shouldMakeContainerAsync);
+            if (dotAwaitContext is DotAwaitContext.AwaitAndConfigureAwait)
+            {
+                displayText += "f";
+                filterText += "F"; // Uppercase F to select "awaitf" if "af" is written.
+            }
+
+            yield return CreateCompletionItem(displayText, filterText, falseKeyword, completionProperties);
+
+            static CompletionItem CreateCompletionItem(string displayText, string filterText, string falseKeyword, ImmutableDictionary<string, string> completionProperties)
+            {
+                var makeContainerAsync = completionProperties.ContainsKey(AwaitCompletionChange.MakeContainerAsync);
```

Comment:
do you need a flag for this?  can't this be determined at the point we're inserting the item.